library/Stratosphere/ResourceProperties.hs

```diff
@@ -0,0 +1,26 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Stratosphere.ResourceProperties
+  ( ResourceProperties(..)
+  , ToResourceProperties(..)
+  , resourcePropertiesJSON
+  ) where
+
+import Data.Aeson
+import Data.Aeson.Types (Pair)
+import Data.Text (Text)
+
+data ResourceProperties
+  = ResourceProperties
+  { resourcePropertiesType :: Text
+  , resourcePropertiesProperties :: Value -- Maybe use Object here?
```

Comment:
`Value` is fine, best-case scenario with `Object` is that you have to explode if something doesn't return a `HashMap`. Also remember `-XStrictData` or `!`s :stuck_out_tongue:

Reply:
Both good points!

Reply:
I might just get rid of the `ToJSON` instance for each resource so I don't wrap them up in `Value`.

Reply:
Okay I made this `Object`, inlined `ToJSON` for any resources into the `ToResourceProperties` instance, and I also got rid of `TeResourceProperties` instances for resource property types (which aren't resources and shouldn't be able to become resources).